graphql/schemabuilder/pagination.go

```diff
@@ -51,14 +51,31 @@ type PaginationArgs struct {
 	Before *string
 }
 
+func (p PaginationArgs) Limit() int {
+	if p.First != nil {
+		return int(*p.First)
+	}
+	if p.Last != nil {
+		return int(*p.Last)
+	}
+	return 0
+}
+
 // PaginationInfo can be returned in a PaginateFieldFunc. The TotalCount function returns the
 // totalCount field on the connection Type. If the resolver makes a SQL Query, then HasNextPage and
 // HasPrevPage can be resolved in an efficient manner by requesting first/last:n + 1 items in the
 // query. Then the flags can be filled in by checking the result size.
 type PaginationInfo struct {
-	TotalCount  func() int64
-	HasNextPage bool
-	HasPrevPage bool
+	TotalCountFunc func() int64
+	HasNextPage    bool
+	HasPrevPage    bool
+}
+
+func (i PaginationInfo) TotalCount() int64 {
+	if i.TotalCountFunc == nil {
+		return 0
```

Comment:
can we error in this case? or elsewhere in initialization assert that `TotalCountFunc` is set

Reply:
We can error if it's not set, sure